RTIMULib/IMUDrivers/RTIMUMPU9250.cpp

```diff
@@ -585,14 +585,12 @@ bool RTIMUMPU9250::IMURead()
 
 #else
 
-    if (count > MPU9250_FIFO_CHUNK_SIZE * 40) {
-        // more than 40 samples behind - going too slowly so discard some samples but maintain timestamp correctly
-        while (count >= MPU9250_FIFO_CHUNK_SIZE * 10) {
-            if (!m_settings->HALRead(m_slaveAddr, MPU9250_FIFO_R_W, MPU9250_FIFO_CHUNK_SIZE, fifoData, "Failed to read fifo data"))
-                return false;
-            count -= MPU9250_FIFO_CHUNK_SIZE;
-            m_imuData.timestamp += m_sampleInterval;
-        }
+    // more than 10 samples behind - going too slowly so discard some samples but maintain timestamp correctly
+    while (count >= MPU9250_FIFO_CHUNK_SIZE * 10) {
```

Comment:
Why would you make this dependent on the fifo chunk size? the number of samples to keep should only depend on the filtering you want to run after. Am I missing something?